src/index.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { createStore, applyMiddleware, compose} from 'redux';
+import { Provider } from 'react-redux';
+
+import App from './App';
+import reducer from './reducers';
+import thunk from 'redux-thunk';
+
+import websocketMiddleware from './actions/websocket';
+import { getAllShops } from './actions';
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+const store = createStore(reducer, composeEnhancers(applyMiddleware(thunk, websocketMiddleware('ws://127.0.0.1:8080/'))));
```

Comment:
Actually, it's totally useless... I don't really use it but it was here because I wanted to do something with a rest API initially.
In my actions file, I created a function for each action that can handle more conditions and alternate the value before dispatching the action. All of these "middle" functions should be removed and the action function should be called in the containers instead.